src/server/transport/socketio.ts

```diff
@@ -185,14 +185,14 @@ export class SocketIO {
             );
           }
         });
-        socket.on('chat', async (matchID, chatMessage) => {
+        socket.on('chat', async (matchID, chatMessage, credentials) => {
```

Comment:
Similarly to in the client transport, let’s type this callback using types from `Master`.

```suggestion
        socket.on('chat', async (...args: Parameters<Master['onChatMessage']>) => {
          const [matchID, chatMessage, credentials] = args;
```

These types don’t change how anything behaves or compiles, but help keep things in sync if we ever refactor these interfaces.